### DISCUSSION TITLE
Adding new instrumentation Intellij/Gradle help

### DISCUSSION THREAD

**Discussion by robododge** [open-telemetry/opentelemetry-java-instrumentation#2884] [Q&A] [answered]
Hi,

I'm writing new instrumentation and am struggling when adding the sub directories under instrumentation/ directory.   It seems to be a very fragile process and if Intellij/Gradle doesn't  pick it up the first time, then that instrumentation doesn't stand a chance of getting recognized.

I'm following the [guide here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation.md#folder-structure). but it doesn't seem to address sub directories

It seems if I add something simple like this, it works:
1. create new dir`instrumentation/jetty-http/javaagent`
2. add a grade file with the exact name jetty-http-javaagent.gradle
3. create `src/main/java/` and put a starter java file for the module in io.opentelemetry.javaagent.instrumentation.jettyhttp
4. update settings.gradle to contain `include ':instrumentation:jetty-http:javaagent` (this step will tell intellij to kick in and resolve the new gradle module with proper java bindings)

But, when I go for a sub directory like: `instrumentation/jetty-http/jetty-v8/javaagent` and modify the steps above to match the new sub directory none of the gradle tasks get bound and my java code still shows up as non-compiled text files.

Am I missing some naming convention or setup step to create a new instrumentation module with a sub-directory structure?

**Comment by robododge** [comment, marked as answer]
I found out the cause.

Gradle modules using the provided gradle file `apply from: "$rootDir/gradle/instrumentation.gradle` must follow a naming convention for sub directories.

I my case, if I change the name to `':instrumentation:jetty-http:jetty-http-9.0:javaagent'` it will work because the sub-directory jetty-http-9.0 is prefixed with jetty-http

**Reply by trask** [reply]
@robododge glad you figured this out! I opened an issue to make sure that we improve the doc to cover this #2891

**Reply by robododge** [reply]
Great, I knew there was probably some inherit naming standards, thanks trask!
